go.mod

```diff
@@ -1,8 +1,8 @@
 module sigs.k8s.io/cluster-api-provider-azure
 
-go 1.22.7
```

Comment:
CAPZ updates its go minor version by absorbing CAPI's minor releases. So we will hold this PR for now. 
/hold